index.ts

```diff
@@ -289,34 +321,41 @@ export function reverse ({
  * @param {string} [$0.url]                     optional URL to override ESRI reverseGeocode endpoint
  * @return {Promise}                            A Promise that'll get resolved with search result
  */
-export function search ({
+export function search({
+  boundary,
   clientId,
   clientSecret,
-  boundary,
   focusPoint,
   forStorage = false,
   magicKey,
   size = 10,
   text,
   url
 }: BaseQuery & {
-  boundary?: Boundary,
-  focusPoint?: any,
-  forStorage?: boolean,
-  magicKey?: string,
-  size?: number,
-  text: string,
+  boundary?: Boundary
+  focusPoint?: Point
+  forStorage?: boolean
+  magicKey?: string
+  size?: number
+  text?: string
 }): Promise<BaseResponse> {
```

Comment:
The `BaseResponse` is not even correct here as the query part is an Object with the text key/value. Please update the return type.

Reply:
Fixed in cdd90839c90a85888732fb6718964b8ddd8fb1b6